### PR TITLE
Migrate frint-cli to TypeScript

### PR DESCRIPTION
**Pull request by markvincze**
This is a starting point to move `frint` to TypeScript.
I started with `frint-cli`, because it's less critical, and it's self-contained. If this looks good and we agree to start moving to TS, then the core `frint` package would be a good next candidate.

Although `frint-cli` is really simple and its code itself is not really utilizing TS, the tooling is already valuable. For example these are producing errors at compile time without running any tests, while when using JS, we don't see anything.

image

image

And we get proper autocomplete too out of the box, and we don't have to maintain `d.ts` files manually:

image

The exciting part is going to be migrating `frint`, because there we'll have to make interesting decisions about how to design the TS type model.

### CODE REVIEW

File: packages/frint-cli/package.json [hunk 1]
```diff
@@ -2,15 +2,14 @@
...
-    "frint": "./bin/frint.js"
+    "frint": "./lib/bin/frint.js"
   },
   "scripts": {
-    "lint": "cross-env ../../node_modules/.bin/eslint --color '{root,commands,bin}/**/*.js'",
-    "transpile": "echo \"Nothing to transpile\"",
-    "test": "cross-env ../../node_modules/.bin/mocha --colors --recursive ./commands/*.spec.js",
+    "lint": "cross-env ./node_modules/.bin/tslint -p .",
+    "transpile": "cross-env ./node_modules/.bin/tsc",
+    "test": "cross-env ../../node_modules/.bin/mocha --require ts-node/register --colors --recursive ./src/**/*.spec.ts",
```
Comment (markvincze) [inline packages/frint-cli/package.json]: There are multiple ways to run TS unit tests.  
We could also simply `transpile` first and then run `mocha` on the transpiled JS, but this option to use `ts-node` seems to be more popular (and this way `watch` can work properly too).

File: package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "description": "Framework for building front-end apps",
   "private": true,
   "scripts": {
-    "bootstrap": "lerna bootstrap",
+    "bootstrap": "lerna bootstrap --hoist",
```
Comment (markvincze) [inline package.json]: The `--hoist` flag is needed due to https://github.com/Microsoft/TypeScript/issues/20945

**Comment by markvincze** [comment]
The CI build is failing due to linting, I'll check that when I'll have time.

**Comment by fahad19** [comment]
💖💖💖

Many thanks for leading this TypeScript initiative, @markvincze!!!

(Will check this PR out in more details toorrow)

**Comment by discosultan** [comment]
merge

**Comment by fahad19** [comment]
Refs #388

**Pull request merged**